app/cdap/components/PipelineDiff/DiffInfo/DiffInfoTable.tsx

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React from 'react';
+import styled from 'styled-components';
+
+import { DiffIndicator, IPipelineStage } from '../types';
+import Table from '@material-ui/core/Table';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import TableCell from '@material-ui/core/TableCell';
+import TableContainer from '@material-ui/core/TableContainer';
+
+const DiffInfoTableContainer = styled(TableContainer)`
+  width: 100%;
+  flex-grow: 1;
+`;
+
+function getEntryData(diffIndicator: DiffIndicator, propertyName: string, propertyValue) {
+  if (diffIndicator === DiffIndicator.DELETED || diffIndicator === DiffIndicator.ADDED) {
+    return { name: propertyName, value: propertyValue as string };
+  }
+  if (typeof propertyValue === 'string' && propertyName.match(/__added$/)) {
+    return {
+      name: propertyName.replace(/__added$/, ''),
+      oldValue: '-',
+      newValue: propertyValue,
+    };
+  } else if (typeof propertyValue === 'string' && propertyName.match(/__deleted$/)) {
+    return {
+      name: propertyName.replace(/__deleted$/, ''),
```

Comment:
these regex's should probably be constants

Reply:
done